### ISSUE TITLE
support for codegen-ing common scripts

### ISSUE THREAD

**Issue by oyvindberg** [oyvindberg/bleep#224]
- publish
- native image
- anything else?

**Comment by sideeffffect** [comment]
Isn't is better to have copy-paste friendly snippets in documentation?

**Comment by oyvindberg** [comment]
Not sure!

There is some precedence for generating code on behalf of the user already in bleep. 
- `bleep build new ...` generates a new build
- multiple commands under `bleep build ...` updates your build
-  `bleep import` generates a script (alongside your new build) to generate resources. 
 
I think it's quite smooth as long as it tells you what it's doing, which files and so on.

One point is that these snippets typically will need dependencies in the build to work, which is one more thing to document and copy/paste

do you have a case in mind where putting it in documentation is clearer?

**Comment by sideeffffect** [comment]
> do you have a case in mind where putting it in documentation is clearer?

Not really, to be hones. I guess I have this intuition about generating code that it is brittle and doesn't scale well. But that's just me. Don't let me stop you :wink:

**Comment by oyvindberg** [comment]
I'm a mostly a [fan](https://github.com/ScalablyTyped/Distribution) of code generation :D

**Comment by sideeffffect** [comment]
Oh, in that sense (ScalablyTyped, ScalaPB, Guardrail, ...), I am too :100: 
But this felt a little bit different to me.
